sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -46,35 +47,42 @@
 public class ReactiveCosmosTemplate implements ReactiveCosmosOperations, ApplicationContextAware {
 
     private final MappingCosmosConverter mappingCosmosConverter;
-    private final String databaseName;
-
+    private final CosmosFactory cosmosFactory;
+    private final CosmosConfig cosmosConfig;
     private final CosmosAsyncClient cosmosAsyncClient;
-    private final ResponseDiagnosticsProcessor responseDiagnosticsProcessor;
-    private final boolean isPopulateQueryMetrics;
```

Comment:
Same here, please keep the variables, and populate them in the constructor. 
Also, I missed it in my earlier PR, but can you please rename `isPopulateQueryMetrics` -> `enableQueryMetrics`

Reply:
I delete these variables in ReactiveCosmosTemplate, we can get these variables from CosmosConfig variable directly.

Reply:
Yes, you are right, but the reason I am suggesting to bring back these variables is because of two reasons. 
1. It will improve the performance of the application - since these are final variables, they will be only calculated once and JVM will optimize the runtime code by replacing their actual values. 
2. It will be much easier to read the code that way - as it is much simpler to read the instance variables as compared to calling APIs. 

```
    private final String databaseName;
    private final ResponseDiagnosticsProcessor responseDiagnosticsProcessor;
    private final boolean enableQueryMetrics;
```

What are your thoughts ?

Reply:
I delete these variables before because I think these variables can be got from CosmosConfig or CosmosFactory, so there is no need to define the variables again, it's more clear. and JVM won't affect a lot of performance. But I think your consider is good, I'm not very familiar with it before. Now I bring back these variables